### PR TITLE
Pod fail minor fixes

### PR DESCRIPTION
**Pull request by sonalk0209**
# Problem Statement
Minor fixes for data pod failure file

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by sonalk0209** [comment]
[collect_pr1465.log](https://github.com/Seagate/cortx-test/files/8553798/collect_pr1465.log)
Collection log

File: tests/ha/test_ha_pod_restart.py [hunk 1]
```diff
@@ -2751,6 +2751,68 @@ def test_server_pod_restart_kubectl_delete(self):
...
+
+        LOGGER.info("Deleting pod %s", pod_name)
+        resp = self.node_master_list[0].delete_pod(pod_name=pod_name, force=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to delete pod {pod_name} by kubectl delete")
+        LOGGER.info("Step 2: Successfully shutdown/deleted pod %s by kubectl delete", pod_name)
+
+        LOGGER.info("Waiting for %s seconds before taking cluster status",
+                    HA_CFG["common_params"]["20sec_delay"])
+        time.sleep(HA_CFG["common_params"]["20sec_delay"])
```
Comment (ApurwaMohite) [inline tests/ha/test_ha_pod_restart.py:2791]: By default grace-period for pod deletion is 30 seconds.

File: tests/ha/test_ha_pod_restart.py [hunk 2]
```diff
@@ -2751,6 +2751,68 @@ def test_server_pod_restart_kubectl_delete(self):
...
+    @CTFailOn(error_handler)
+    def test_pod_shutdown_kubectl_delete(self):
+        """
+        Verify IOs before and after data pod failure; pod shutdown deleting pod
+        using kubectl delete.
+        """
+        LOGGER.info("STARTED: Verify IOs before and after data pod failure, "
+                    "pod shutdown by deleting pod using kubectl delete.")
+
+        LOGGER.info("STEP 1: Create s3 account and perform WRITEs-READs-Verify-DELETEs with "
```
Comment (ApurwaMohite) [inline tests/ha/test_ha_pod_restart.py:2767]: Create IAM user*

File: tests/ha/test_ha_pod_restart.py [hunk 3]
```diff
@@ -2751,6 +2751,68 @@ def test_server_pod_restart_kubectl_delete(self):
...
+        resp = self.node_master_list[0].delete_pod(pod_name=pod_name, force=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to delete pod {pod_name} by kubectl delete")
+        LOGGER.info("Step 2: Successfully shutdown/deleted pod %s by kubectl delete", pod_name)
+
+        LOGGER.info("Waiting for %s seconds before taking cluster status",
+                    HA_CFG["common_params"]["20sec_delay"])
+        time.sleep(HA_CFG["common_params"]["20sec_delay"])
+        LOGGER.info("Step 3: Check cluster status")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
```
Comment (ApurwaMohite) [inline tests/ha/test_ha_pod_restart.py:2793]: May be we can poll here for 1min, to allow new pod creation.

File: tests/ha/test_ha_pod_restart.py [hunk 4]
```diff
@@ -2751,6 +2751,68 @@ def test_server_pod_restart_kubectl_delete(self):
...
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 3: Cluster is in degraded state")
+
+        LOGGER.info("Step 5: Check services status on all pods", pod_list)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=pod_list, fail=False)
```
Comment (ApurwaMohite) [inline tests/ha/test_ha_pod_restart.py:2798]: This check is not really needed as check_cluster_status takes care of services check also.

Comment (sonalk0209) [reply tests/ha/test_ha_pod_restart.py:2798]: agree

**Review by ApurwaMohite — APPROVED** [review]

File: tests/ha/test_ha_pod_restart.py [hunk 5]
```diff
@@ -2751,6 +2751,58 @@ def test_server_pod_restart_kubectl_delete(self):
...
+        LOGGER.info("Deleting pod %s", pod_name)
+        resp = self.node_master_list[0].delete_pod(pod_name=pod_name, force=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to delete pod {pod_name} by kubectl delete")
+        LOGGER.info("Step 2: Successfully shutdown/deleted pod %s by kubectl delete", pod_name)
+
+        LOGGER.info("Step 3: Check cluster status")
+        resp = self.ha_obj.poll_cluster_status(self.node_master_list[0], timeout=60)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 3: Cluster is in healthy state.")
```
Comment (rahul-hatwar) [inline tests/ha/test_ha_pod_restart.py:2791]: Are we expecting cluster will be Healthy after kubectl delete?

Comment (sonalk0209) [reply tests/ha/test_ha_pod_restart.py:2791]: yes as kubectl delete restart the pod

File: tests/ha/test_ha_pod_restart.py [hunk 6]
```diff
@@ -2751,6 +2751,68 @@ def test_server_pod_restart_kubectl_delete(self):
...
+
+        LOGGER.info("Waiting for %s seconds before taking cluster status",
+                    HA_CFG["common_params"]["20sec_delay"])
+        time.sleep(HA_CFG["common_params"]["20sec_delay"])
+        LOGGER.info("Step 3: Check cluster status")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 3: Cluster is in degraded state")
+
+        LOGGER.info("Step 5: Check services status on all pods", pod_list)
```
[the remark this answers is not in the document]
Comment (sonalk0209) [reply tests/ha/test_ha_pod_restart.py]: removed

[the remark this answers is not in the document]
Comment (sonalk0209) [reply tests/ha/test_ha_pod_restart.py]: removed the step as cluster check will check all services as well

**Review by rahul-hatwar — APPROVED** [review]
Minor comment. LGTM approved.

**Review by ankita-93 — APPROVED** [review]
LGTM

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**